File: Hashmap/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hmap;
        for(int i = 0; i < nums.size(); ++i) {
            if(hmap.find(target-nums[i]) != hmap.end()) 
                return {hmap[target-nums[i]], i};
            hmap[nums[i]] = i;
        }
        
        return {-1,-1};
    }
};

File: Hashmap/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap();
        int n = nums.length;
        for(int i = 0; i < n; ++i) {
            if(numMap.containsKey(target-nums[i])) 
                return new int[] {numMap.get(target-nums[i]), i};
            numMap.put(nums[i],i);
        }
        throw new IllegalArgumentException("no solution");
    }
}
